File: week7/lab_py/bank/account.py
class Account:
    def __init__(self, account_type):
        self.type = account_type
        self.balance = self.read_balance()

    def read_balance(self):
        print(f'Initial {self.type} balance $', end='')
        return float(input())

    def has_balance(self, amount):
        return self.balance >= amount
    
    def has_type(self,account_type):
        return self.type == account_type

    def deposit(self, amount):
        self.balance += amount

    def withdraw(self, amount):
        self.balance -= amount

    def pay_to(self, amount, target):
        self.balance -= amount
        target.balance += amount

    def __str__(self):
        return f'{self.type} balance: ${self.balance:.2f}'

File: week9/lab_java/bank/Account.java
package bank;

import java.io.Serializable;

public class Account implements Serializable {
    private String type;
    private double balance;

    public Account(String type) {
        this.type = type;
        this.balance = readBalance();        
    }

    private double readBalance() {
        System.out.print("Initial " + this.type + " balance $");
        return Config.IN.nextDouble();
    }

    public boolean hasType(String type){
        return this.type.equals(type);
    }

    public boolean has(double amount) {
        return balance >= amount;
    }
    
    public void deposit(double amount) {
        this.balance += amount;
    }

    public void withdraw(double amount) {
        this.balance -= amount;
    }

    public void transfer(double amount, Account target) {
        this.balance -= amount;
        target.balance += amount;
    }

    @Override
    public String toString() {
        return String.format("%s account has $%.2f", type, balance);
    }         
}

